dashboards-observability/public/components/visualizations/charts/bar/bar.tsx

```diff
@@ -84,7 +84,7 @@ export const Bar = ({ visualizations, layout, config }: any) => {
   const tooltipText = tooltipOptions.tooltipText !== undefined ? tooltipOptions.tooltipText : 'all';
   const barWidth = 1 - (chartStyles.barWidth || barwidth);
   const groupWidth = 1 - (chartStyles.groupWidth || groupwidth);
-  const showLegend = legend.showLegend ?? showlegend;
+  const showLegend = !(legend.showLegend && legend.showLegend !== showlegend);
```

Comment:
I think I've mentioned this in our meetings, instead of having to check if showLegend equals 'show' or 'hidden', it should be more straitforward to have boolean value like showlegend = true/false

Reply:
@mengweieric Updated the code, please have a look.